File: app/src/main/java/org/xdty/callerinfo/contract/PhoneStateContract.java
package org.xdty.callerinfo.contract;

import android.content.Context;

import org.xdty.phone.number.model.INumber;

public interface PhoneStateContract {

    interface View {

        void show(INumber number);

        void showFailed(boolean isOnline);

        void showSearching();

        void hide(String number);

        void close(String number);

        boolean isShowing();

        Context getContext();

        void showMark(String number);
    }

    interface Presenter extends BasePresenter {

        boolean matchIgnore(String number);

        void handleRinging(String number);

        void handleOffHook(String number);

        void handleIdle(String number);

        void resetCallRecord();

        boolean checkClose(String number);

        boolean isIncoming(String number);

        void saveInCall();

        boolean isRingOnce();

        void searchNumber(String number);

        void setOutGoingNumber(String number);

        boolean canReadPhoneState();
    }
}

File: app/src/main/java/org/xdty/callerinfo/contract/PhoneStateContract.kt
package org.xdty.callerinfo.contract

import android.content.Context
import org.xdty.phone.number.model.INumber

interface PhoneStateContract {
    interface View {
        fun show(number: INumber)
        fun showFailed(isOnline: Boolean)
        fun showSearching()
        fun hide(number: String)
        fun close(number: String)
        val isShowing: Boolean
        val context: Context
        fun showMark(number: String)
    }

    interface Presenter : BasePresenter {
        fun matchIgnore(number: String): Boolean
        fun handleRinging(number: String)
        fun handleOffHook(number: String)
        fun handleIdle(number: String)
        fun resetCallRecord()
        fun checkClose(number: String): Boolean
        fun isIncoming(number: String): Boolean
        fun saveInCall()
        val isRingOnce: Boolean
        fun searchNumber(number: String)
        fun setOutGoingNumber(number: String)
        fun canReadPhoneState(): Boolean
    }
}